### PR TITLE
Modify 11_gay

### PR DESCRIPTION
**Pull request by lanpay-lulu**
Implement a working version of gan based on the original 11_gan.py.

Main modifications are as follows:

- G output range from 0 to 255.
- Activation function changes from relu to tanh.
- G objective and D objective change.
- training two step of G and one step of D in one batch.

### CODE REVIEW

**Comment by lanpay-lulu** [comment]
Why the Travis CI build failed？
Any help?

**Comment by hunkim** [comment]
I'll look into it.

File: 11_gan.py [hunk 1]
```diff
@@ -37,40 +38,74 @@ def xavier_initializer(shape):
...
+    h2 = tf.matmul(h1, w['out']) + w['b2']
+    return h2 # use h2 to calculate logits loss
 
 def generate_z(n=1):
     return np.random.normal(size=(n, z_size))
 
 sample = G(z)
 
 # Objective functions
-G_objective = -tf.reduce_mean(tf.log(D(G(z))))
```
Comment (hunkim) [inline 11_gan.py]: This code is more readable. Why we need to use cross_entropy?

**Comment by hunkim** [comment]
Overall, I like to have simple examples. Do you think you can simplify your PR and make only necessary changes?

File: 11_gan.py (hunk 1 above)
Comment (lanpay-lulu) [inline 11_gan.py]: When D(G(z))==0, log(0) will cause exception.

**Comment by hunkim** [comment]
Add `+1e-9' or something?

Sung

**Comment by lanpay-lulu** [comment]
I still insist this is the simplest gan version. And tf.nn.sigmoid_cross_entropy_with_logits is the most common function in tf that everyone need to know, and to use. This is the version that really generate something meaningful. That`s why I strongly recommend it.  ^ ^

File: 11_gan.py [hunk 2]
```diff
@@ -37,40 +38,74 @@ def xavier_initializer(shape):
...
-D_objective = -tf.reduce_mean(tf.log(D(X)) + tf.log(1 - D(G(z))))
+def sigmoid_cross_entropy_with_logits(x, y):
```
Comment (hunkim) [inline 11_gan.py]: Why did you add the wrapper? Why not just using sigmoid_cross_entropy_with_logits?

Comment (lanpay-lulu) [inline 11_gan.py]: The tf 1.0 api is as follows:
def sigmoid_cross_entropy_with_logits(_sentinel=None, labels=None, logits=None, name=None)
Using this wrapper can let us ignore the param names such as logics and labels(targets in lower version).

Comment (hunkim) [inline 11_gan.py]: I know, but this wrapper add another complexity. :-)

Comment (lanpay-lulu) [inline 11_gan.py]: Ok. It make sense. I`ll update it later. Any more comments?

**Comment by hunkim** [comment]
D out is 1, right? In this case, could you use just logistic loss? (not cross entropy?)

**Comment by hunkim** [comment]
Anyway, I am ready to merge this. Do you have anything to push in this change?

**Comment by lanpay-lulu** [comment]
That`s all to merge for this version. We can continue to upgrade it in the future. ^ ^

**Comment by hunkim** [comment]
Sounds good.
